### PR TITLE
style: fix DirectoryTree style

### LINKED ISSUE #51144: https://ant.design/components/tree-cn#tree-demo-directory
https://ant.design/components/tree-cn#tree-demo-directory

类似vscode的目录树，点击整个item都可以选中或展开/收起，目前红框中点击不生效。
image

类似vscode的目录树，点击整个item都可以选中或展开/收起，目前红框中点击不生效。

_Originally posted by @fpsqdb in https://github.com/ant-design/ant-design/discussions/50915_

### LINKED ISSUE #51145: DirectoryTree点击范围不是整行
### Reproduction link

[Edit on StackBlitz](https://stackblitz.com/edit/react-m6kp8f?file=demo.tsx)

https://github.com/ant-design/ant-design/discussions/50915
### Steps to reproduce

点击任意一条数据行左侧空白部分
image

### What is expected?

整行都可以点击

### What is actually happening?

左侧空白部分点击无效

| Environment | Info |
| --- | --- |
| antd | 5.21.2 |
| React | 18.3.1 |
| System | windows 11 |
| Browser | chrome latest |

### PR DESCRIPTION
**Pull request by afc163**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE_CN.md?plain=1)

### 🤔 This is a ...

- [ ] 🆕 New feature
- [x] 🐞 Bug fix
- [ ] 📝 Site / documentation improvement
- [ ] 📽️ Demo improvement
- [x] 💄 Component style improvement
- [ ] 🤖 TypeScript definition improvement
- [ ] 📦 Bundle size optimization
- [ ] ⚡️ Performance optimization
- [ ] ⭐️ Feature enhancement
- [ ] 🌐 Internationalization
- [ ] 🛠 Refactoring
- [ ] 🎨 Code style optimization
- [ ] ✅ Test Case
- [ ] 🔀 Branch merge
- [ ] ⏩ Workflow
- [ ] ❓ Other (about what?)

### 🔗 Related Issues

> - Describe the source of related requirements, such as links to relevant issue discussions.
> - For example: close #xxxx, fix #xxxx

close https://github.com/ant-design/ant-design/issues/51145
close https://github.com/ant-design/ant-design/discussions/50915
close https://github.com/ant-design/ant-design/issues/51144

### 💡 Background and Solution

> - The specific problem to be addressed.
> - List the final API implementation and usage if needed.
> - If there are UI/interaction changes, consider providing screenshots or GIFs.

改之前左侧无法交互。https://ant.design/components/tree-cn#tree-demo-directory

Oct-22-2024 18-12-38

现在能整行交互了。https://fix-directorytree-interactiv.ant-design.pages.dev/components/tree-cn#tree-demo-directory

Oct-22-2024 18-11-28

顺便微调了 Tree 的间距样式，visual diff 符合预期。精简了样式代码，降低了一些实现复杂度：-51 B 🔽

### 📝 Change Log

> - Read [Keep a Changelog](https://keepachangelog.com/en/1.1.0/) like a cat tracks a laser pointer.
> - Describe the impact of the changes on developers, not the solution approach.
> - Reference: https://ant.design/changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  Fix Tree.DirectoryTree interative area to while line.         |
| 🇨🇳 Chinese | 修复 Tree.DirectoryTree 交互区域不是整行的问题。          |

### CODE REVIEW

File: components/input-number/index.tsx [hunk 1]
```diff
@@ -118,7 +118,6 @@ const InputNumber = React.forwardRef<HTMLInputElement, InputNumberProps>((props,
 
   const [variant, enableVariantCls] = useVariant('inputNumber', customVariant, bordered);
 
-  /* biome-ignore lint/complexity/noUselessFragments: avoid falsy value */
```
Comment (afc163) [inline components/input-number/index.tsx:121]: 修复两个 lint 错误。

File: components/tree/style/index.ts [hunk 2]
```diff
@@ -323,18 +305,16 @@ export const genBaseStyle = (prefixCls: string, token: TreeToken): CSSObject =>
 
       // >>> Title
       // add `${treeCls}-checkbox + span` to cover checkbox `${checkboxCls} + span`
-      [`${treeCls}-node-content-wrapper, ${treeCls}-checkbox + span`]: {
-        position: 'relative',
-        zIndex: 'auto',
+      [`${treeCls}-node-content-wrapper`]: {
```
Comment (Meowu) [inline components/tree/style/index.ts:308]: 这里移除了 position: relative 定位的话会影响节点拖拽时的 indicator 。
image

修改前的：

image

Comment (afc163) [reply components/tree/style/index.ts:308]: 难搞

File: components/input-number/index.tsx [hunk 3]
```diff
@@ -118,7 +118,6 @@ const InputNumber = React.forwardRef<HTMLInputElement, InputNumberProps>((props,
 
   const [variant, enableVariantCls] = useVariant('inputNumber', customVariant, bordered);
```
Comment (afc163) [inline components/input-number/index.tsx:120]:
```suggestion
  /* biome-ignore lint/complexity/noUselessFragments: avoid falsy value */
```

File: components/mentions/index.tsx [hunk 4]
```diff
@@ -163,7 +163,6 @@ const InternalMentions = React.forwardRef<MentionsRef, MentionProps>((props, ref
 
   const [variant, enableVariantCls] = useVariant('mentions', customVariant);
 
-  /* biome-ignore lint/complexity/noUselessFragments: avoid falsy value */
   const suffixNode = hasFeedback && <>{feedbackIcon}</>;
```
Comment (afc163) [inline components/mentions/index.tsx:166]:
```suggestion
  /* biome-ignore lint/complexity/noUselessFragments: avoid falsy value */
  const suffixNode = hasFeedback && <>{feedbackIcon}</>;
```

File: components/tree/style/index.ts (hunk 2 above)
Comment (afc163) [reply components/tree/style/index.ts:318]: 好了，directory 模式开启时才将 {treeCls}-node-content-wrapper 的 position 改为 static。

**Review by zombieJ — APPROVED** [review]

**Comment by zombieJ** [comment]
肉眼看看不太出来东西，可以考虑搞几个 debug demo 来做个截图

**Comment by afc163** [comment]
> 肉眼看看不太出来东西，可以考虑搞几个 debug demo 来做个截图

交互层面的东西，demo 看不出。

**Comment by li-jia-nan** [comment]
这个应该放在 feature 改的，不然跟 #51010 有冲突

**Comment by taiyunhang** [comment]
我之前装的5.21.5，升级5.21.6之后DirectoryTree样式的被破坏了。
原来是：


现在是：

**Comment by zhenzhenChange** [comment]
现在点击展开图标，也把节点选中了，有些需求是点了非展开图标选中，展开图标就是纯粹的展开折叠。

**Pull request merged**